Gems/Pointcloud/Code/Source/Render/PointcloudFeatureProcessor.cpp

```diff
@@ -0,0 +1,265 @@
+        m_shader = AZ::RPI::LoadCriticalShader(shaderFilePath);
+
+        if (!m_shader)
+        {
+            AZ_Error("PointcloudFeatureProcessor", false, "Failed to load required stars shader.");
+            return;
+        }
+        AZ::Data::AssetBus::Handler::BusConnect(m_shader->GetAssetId());
+
+        auto drawSrgLayout = m_shader->GetAsset()->GetDrawSrgLayout(m_shader->GetSupervariantIndex());
+        AZ_Error(
+            "PointcloudFeatureProcessor", drawSrgLayout, "Failed to get the draw shader resource group layout for the pointcloud shader.");
+        if (drawSrgLayout)
+        {
+            m_drawSrg =
+                AZ::RPI::ShaderResourceGroup::Create(m_shader->GetAsset(), m_shader->GetSupervariantIndex(), drawSrgLayout->GetName());
+            m_pointSizeIndex.Reset();
+            m_modelMatrixIndex.Reset();
+        }
+
+        m_drawListTag = m_shader->GetDrawListTag();
+
+        auto viewportContextInterface = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
+        AZ_Assert(viewportContextInterface, "PointcloudFeatureProcessor requires the ViewportContextRequestsInterface.");
+        auto viewportContext = viewportContextInterface->GetViewportContextByScene(GetParentScene());
+        AZ_Assert(viewportContext, "PointcloudFeatureProcessor requires a valid ViewportContext.");
+        m_viewportSize = viewportContext->GetViewportSize();
+
+        EnableSceneNotification();
+
+        AZ::RPI::ViewportContextIdNotificationBus::Handler::BusConnect(viewportContext->GetId());
+    }
+
+    PointcloudFeatureProcessorInterface::PointcloudHandle PointcloudFeatureProcessor::AcquirePointcloud(
+        const AZStd::vector<PointcloudAsset::CloudVertex>& cloudVertexData)
+    {
+        m_currentPointcloudDataIndex++;
+        auto& pcData = m_pointcloudData[m_currentPointcloudDataIndex];
+        pcData.m_index = m_currentPointcloudDataIndex;
+        const uint32_t elementSize = sizeof(PointcloudAsset::CloudVertex);
+        const uint32_t elementCount = static_cast<uint32_t>(cloudVertexData.size());
+        const uint32_t bufferSize = elementCount * elementSize; // bytecount
+
+        AZ_Printf("PointcloudFeatureProcessor", "PointcloudFeatureProcessor SetCloud %d, bytesize %d", elementCount, bufferSize);
+        pcData.m_pointData = cloudVertexData;
+        pcData.m_vertices = elementCount;
+
+        AZ::RPI::CommonBufferDescriptor desc;
+        desc.m_poolType = AZ::RPI::CommonBufferPoolType::ReadWrite;
+        desc.m_bufferName = AZStd::string::format("PointcloudFeatureProcessor, %d", pcData.m_index);
+        desc.m_byteCount = bufferSize;
+        desc.m_elementSize = elementSize;
+        desc.m_bufferData = pcData.m_pointData.data();
+
+        AZStd::vector<uint8_t> bufferData;
+
+        bufferData.resize_no_construct(desc.m_byteCount);
+        memcpy(bufferData.data(), desc.m_bufferData, desc.m_byteCount);
```

Comment:
this copy of the buffer dated does not seem to be used anywhere.

Reply:
Oh, my bad, It was used in debugging - I am removing this code, thanks for pointing it up!